NDTensors/src/NDTensors.jl

```diff
@@ -220,13 +223,31 @@ function disable_tblis()
   return nothing
 end
 
+function backend_octavian()
+  throw(UndefVarError)
+end
+
 function __init__()
-  @require TBLIS = "48530278-0828-4a49-9772-0f3830dfa1e9" begin
-    enable_tblis()
-    include("tensoralgebra/tblis.jl")
-  end
-  @require Octavian = "6fd5a793-0b7e-452c-907f-f8bfe9c57db4" begin
-    include("linearalgebra/octavian.jl")
+  @static if !isdefined(Base, :get_extension)
+    @require TBLIS = "48530278-0828-4a49-9772-0f3830dfa1e9" begin
+      enable_tblis()
+      include("../ext/NDTensorTBLIS/NDTensorTBLIS.jl")
+    end
+    @require Octavian = "6fd5a793-0b7e-452c-907f-f8bfe9c57db4" begin
+      #include("linearalgebra/octavian.jl")
+      include("../ext/NDTensorOctavian/NDTensorOctavian.jl")
+    end
+
+    @require CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba" begin
+      if CUDA.functional()
+        include("../ext/NDTensorCUDA/NDTensorCUDA.jl")
+      end
+    end
+
+    @require Metal = "dde4c033-4e86-420c-a63e-0dd931031962" begin
+      println("NDTensors is compiled with Metal.jl")
```

Comment:
Print statement should be removed.

Reply:
I didn't know if we wanted to warn/inform users that they instantiated extensions when they add certain packages.